### PR TITLE
PBI 223 : sort vendor by rating

### PR DESCRIPTION
**Pull request by bryannaufal**
## Describe your changes
Implementing sort vendor by rating in vendor page (vendor table)

## Issue ticket number and link
https://linear.app/adudu/issue/ADU-223/sort-vendor-by-rating

Before sort
image

After sort
image

### CODE REVIEW

File: src/routes/vendor.tsx [hunk 1]
```diff
@@ -22,8 +22,15 @@ export default function VendorPage() {
   const [productFilter, setProductFilter] = useState('')
   const [locationFilter, setLocationFilter] = useState('')
   const { setShowLoadingOverlay } = useCommonStore()
+  const [sortBy, setSortBy] = useState<'name' | 'area_group_name' | 'rating' | ''>('');
+  const [sortOrder, setSortOrder] = useState<'asc' | 'desc' | ''>('');
```
Comment (valenciusap17) [inline src/routes/vendor.tsx]: the default form backend is 'asc' cmiiw, so i think we can replace default value '' as 'asc'

**Comment by edutjie** [comment]
This PR also implemented the sorting logic right? I don't find where is the logic to sort the data apart from just altering the arrow UI. Can you show me the reference of the logic I'm looking for?

File: src/routes/vendor.tsx (hunk 1 above)
Comment (bryannaufal) [reply src/routes/vendor.tsx]: Done

File: src/routes/vendor.tsx [hunk 2]
```diff
@@ -22,8 +22,15 @@ export default function VendorPage() {
...
+  const [sortBy, setSortBy] = useState<'name' | 'area_group_name' | 'rating' | ''>('');
+  const [sortOrder, setSortOrder] = useState<'asc' | 'desc'>('asc');
+
   const { vendors, metadata, isSuccess, error } = useGetVendorsQuery(
-    { page, product: productFilter, location: locationFilter },
+    { page, 
+      product: productFilter, 
+      location: locationFilter, 
+      order_by: sortBy, 
+      order: sortOrder },
```
Comment (bryannaufal) [inline src/routes/vendor.tsx:33]: > This PR also implemented the sorting logic right? I don't find where is the logic to sort the data apart from just altering the arrow UI. Can you show me the reference of the logic I'm looking for?

[image](https://github.com/kg-procurement/kg-procurement/blob/main/internal/vendors/accessor.go#L112-L116)

The backend already includes the sorting logic as part of the API implementation. It supports sorting through pagination arguments such as `OrderBy` and `Order`. To enable the sorting functionality, we just need to pass these arguments as part of the API parameters.

**Comment by FreeJ1nG** [comment]
LGTM, merging

**Pull request merged**